manyfews/calculations/gefs.py

```diff
@@ -0,0 +1,201 @@
+            index = cellIndexFinder(
+                latitudeInfo=lats,
+                longitudeInfo=lons,
+                latValue=latValue,
+                lonValue=lonValue,
+            )
+            vWindValue = grb.values[index]
+
+        # Total Precipitation: unit:kg/m2 (accum).
+        elif grb.parameterName == "Total precipitation" and grb.level == 0:
+            lats, lons = grb.latlons()
+            index = cellIndexFinder(
+                latitudeInfo=lats,
+                longitudeInfo=lons,
+                latValue=latValue,
+                lonValue=lonValue,
+            )
+            totalPrecipValue = grb.values[index]
+
+    return RHvalue, maxTempValue, minTempValue, uWindValue, vWindValue, totalPrecipValue
+
+
+def cellIndexFinder(latitudeInfo, longitudeInfo, latValue, lonValue):
+    """
+    This function is developed for finder the index of the specific cell in gefs data.
+
+    :param latitudeInfo: the latitude matrix of the grid.
+    :param longitudeInfo: the longtitude matrix of the grid.
+    :param latValue: the latitude of the specific cell
+    :param lonValue: the longitude of the specific cell
+    :return: the index of the cell location.
+    """
+
+    # find the Index of the specific location in gefs data.
+    latsIndex = np.where(latitudeInfo == latValue)
+    lonsIndex = np.where(longitudeInfo == lonValue)
+    index = (latsIndex[0][0], lonsIndex[1][0])
+
+    return index
+
+
+def dataBaseWriter(dt, forecastDays, backDays, latValue, lonValue):
+    """
+    This function is used to save data from gefs file into Database.
+    ( calculations_noaaforecast table).
+
+    :param dt: Specify time-step in days. ( the interval is every 0.125 day(3 hours))
+    :param forecastDays: Number of Days into the future that the forecast is for.
+    :param backDays: the back days number you want to extract (unit: day) (Maximum = 3 days).
+    :param latValue: the latitude of the specific cell.
+                    (the solution is 0.5 degree. range [-90, 90] with 0.5 interval)
+    :param lonValue: the longtidue of the specific cell.
+                    (the solution is 0.5 degree. range [-90, 90] with 0.5 interval)
+    :return: none
+    """
+
+    # calculate the number of time steps
+    # For example: update interval: 0.25 day = 6 h.
+    #              Number of Days into the future that the forecast is for
+    #              16 days = 4 * 16 = 65 loop steps.
+    #  Therefore, the gefs files are:
+    #  geavg.t00z.pgrb2a.0p50.f006 ---> geavg.t00z.pgrb2a.0p50.f384
+
+    loopRange = int(forecastDays / dt)
+    deltaHour = int(24 * dt)
+
+    downloadDate = datetime.utcnow() - timedelta(
+        days=int(backDays)
+    )  # download 3 days back GEFS data.
+    fileDate = downloadDate.strftime("%Y%m%d")
+    date = datetime.astimezone(downloadDate, tz=timezone(timedelta(hours=0)))
+
+    for i in range(loopRange):
+        forcastHour = deltaHour + i * deltaHour
```

Comment:
Could you rename `forcastHour` to `forecastHour`?

Reply:
accepted and changed.